--- openjdk8-1.8.0.332.orig/make/common/MakeBase.gmk	2022-04-18 02:32:28.000000000 +0200
+++ openjdk8-1.8.0.332/make/common/MakeBase.gmk	2022-04-24 19:10:33.038994029 +0200
@@ -317,10 +317,10 @@
 
 # Emit the scm:id pair to $@
 define GetSourceTips
-	$(CD) $(SRC_ROOT) ; \
-	if [ -d $(SCM_DIR) -a "$(SCM_VERSION)" != "" ] ; then \
+	$(CD) "$(SRC_ROOT)" ; \
+	if [ -d "$(SCM_DIR)" -a "$(SCM_VERSION)" != "" ] ; then \
 	  $(ID_COMMAND) >> $@ ; \
-	elif [ -f $(SCM_TIP_FILENAME) ] ; then \
+	elif [ -f "$(SCM_TIP_FILENAME)" ] ; then \
           $(SCM_TIP_FILECMD) >> $@ ; \
 	fi;
 	$(PRINTF) "\n" >> $@
